init push for convert of unique_id as a list tests for spark

McKnight-42 opened the pull request:
Part of CT-375

### Description
converting unique_id tests to new pytest version
### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-spark next" section.

gshank commented:
That's a good question, and one I don't have the knowledge to answer completely. In general, I think we should be testing the profiles on which we were running tests/integration  (i.e. all of them except spark_session). If particular profiles should not run particular tests, you can the 'skip_profile' mark for those tests.

jtcohen6 commented:
@McKnight-42 Only the `merge` strategy actually makes use of `unique_key`. (`append` doesn't account for duplicates at all, and `insert_overwrite` operates on entire partitions, instead of row-by-row unique identifiers.)

So our tests here need to be configured to use:
- `file_format: delta`
- `incremental_strategy: merge`
- one of the Databricks profiles

jtcohen6 commented:
@McKnight-42 Looks like all of `test_incremental_unique_id.py` passed!

Another basic test failed, with a flaky connection error:
```
('08S01', '[08S01] [Simba][Hardy] (115) Connection failed with error: Bad Status: Unrecognized response with no error message header. Status code: 502 (115) (SQLExecDirectW)')
```
I just triggered the test to rerun in CircleCI.

So I think you're safe to delete the old duplicated versions of the same test, in [`test/integration/incremental_unique_id_test`](https://github.com/dbt-labs/dbt-spark/tree/main/tests/integration/incremental_unique_id_test)

McKnight-42 commented:
> @McKnight-42 Looks like all of `test_incremental_unique_id.py` passed!
> 
> Another basic test failed, with a flaky connection error:
> 
> ```
> ('08S01', '[08S01] [Simba][Hardy] (115) Connection failed with error: Bad Status: Unrecognized response with no error message header. Status code: 502 (115) (SQLExecDirectW)')
> ```
> 
@jtcohen6 looks like the connection test failed again after I added changelog and deleted the files from what I can see might be more connection issues causing them to fail. any thoughts on if this is on ours or circleci end?

jtcohen6 commented:
@McKnight-42 I think this might just be an intermittent connection error :shrug:

I just opened a new issue for better handling those transient errors, based on the connection configs we have available to us: https://github.com/dbt-labs/dbt-spark/issues/326

ChenyuLInx reviewed (approved):
Looks good!
